crm/modules/member/command.inc.php

```diff
@@ -73,12 +73,50 @@ function command_member_add () {
         return crm_url('members&tab=add');
     }
     
+    // Check for duplicate usernames
+    if (!empty($username)) {
+        
+        // Check whether username is in use
+        $test_username = $username;
+        $esc_test_username = mysqli_real_escape_string($db_connect, $test_username);
+        $sql = "SELECT * FROM `user` WHERE `username`='$esc_test_username'";
+        $res = mysqli_query($db_connect, $sql);
+        if (!$res) crm_error(mysqli_error($res));
+        $username_row = mysqli_fetch_assoc($res);
+        if (!$username_row) {
+            $username = $test_username;
+        } else {
+            error_register('Username already in use, please specify a different username');
+            return crm_url('members&tab=add');
```

Comment:
is this the correct thing to do if name already in use?  Does it clear out their fields and make them type everything in again?  It would be best if it could keep everything filled out so they just have to change the one username field.

Reply:
Not sure how to implement that, and was duplicating behaviour already used for other forms